[Yang Xiquan] iP

cindyangXQ opened the pull request:
## DukePro
>“Your mind is for having ideas, not holding them.” – David Allen ([Source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is, 
1. download it from [here](https://github.com/cindyangXQ/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👍 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

cindyangXQ committed:
Level-1

cindyangXQ committed:
Level-2

cindyangXQ committed:
Tasks recorded as Task instead of String

cindyangXQ committed:
Level-3

cindyangXQ committed:
Level-4

cindyangXQ committed:
A-TextUiTesting

cindyangXQ committed:
Level-5

cindyangXQ committed:
Level-6

cindyangXQ committed:
refine oop structure

sharpstorm commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+    public Task(String content, boolean done) {
+        this.content = content;
+        this.done = done;
+    }
```
Should `done` be named differently to indicate that it is a boolean variable? I noticed the same issue in several other classes related to `Task`
```suggestion
    public Task(String content, boolean isDone) {
        this.content = content;
        this.isDone = isDone;
    }
```

sharpstorm commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+        this.printAdd();
+    }
+
+    public void addDdl(String s) {
```
I feel that the parameter name is too short, perhaps consider using a more descriptive parameter name? I noticed this issue in the other `add` methods too.
```suggestion
    public void addDdl(String taskDescription) {
```

sharpstorm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,91 @@
+            + indent + "| | | | | | | |/ / _ \\\n"
+            + indent + "| |_| | |_| |   <  __/\n"
+            + indent + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String byePhrase = "Bye. Hope to see you again soon!";
```
Should constants follow another capitalization convention?
```suggestion
    private static final String SEG_LINE = "    ____________________________________________________________";
    private static final String INDENT = "    ";
    private static final String LOGO = INDENT+ " ____        _        \n"
            + indent + "|  _ \\ _   _| | _____ \n"
            + indent + "| | | | | | | |/ / _ \\\n"
            + indent + "| |_| | |_| |   <  __/\n"
            + indent + "|____/ \\__,_|_|\\_\\___|\n";
    private static final String BYE_PHRASE = "Bye. Hope to see you again soon!";
```

sharpstorm commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.List;
+
+public class TaskList {
+    private static final String indent = "    ";
```
Should this constant follow another capitalization convention? Perhaps you can consider simply using the same `indent` constant declared in the `Duke` class?
```suggestion
    private static final String INDENT = "    ";
```

sharpstorm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,91 @@
+                } else {
+                    throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                }
+            }
+            catch (DukeException e) {
+                System.out.println(indent + e);
+            }
+            finally {
+                System.out.println(segLine);
+            }
```
Should the try-catch block be following 2 different style of brackets?
```suggestion
            } catch (DukeException e) {
                System.out.println(indent + e);
            } finally {
                System.out.println(segLine);
            }
```

yokester98 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+public class Deadline extends Task {
```
Missing Javadoc for all your files! May want to include Javadoc to tell the reader how the code works.

yokester98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,91 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static final String segLine = "    ____________________________________________________________";
```
```suggestion
    private static final String SEGLINE = "    ____________________________________________________________";
```
Constant names must be all uppercase using underscore to separate words.

yokester98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,91 @@
+                    }
+                    tasks.addTodo(s.substring(5));
+                } else if (s.startsWith("deadline")) {
+                    if (s.length() <= 9) {
+                        throw new DukeException("The description of a deadline cannot be empty.");
+                    }
+                    tasks.addDdl(s.substring(9));
+                } else if (s.startsWith("event")) {
+                    if (s.length() <= 6) {
+                        throw new DukeException("The description of an event cannot be empty.");
+                    }
+                    tasks.addEvt(s.substring(6));
+                } else if (s.startsWith("mark")) {
+                    Scanner temp = new Scanner(s.substring(4));
+                    if (temp.hasNextInt()) {
+                        tasks.mark(temp.nextInt() - 1, true);
+                    } else {
+                        throw new DukeException("Please enter an index to mark.");
+                    }
+                } else if (s.startsWith("unmark")) {
+                    Scanner temp = new Scanner(s.substring(6));
+                    if (temp.hasNextInt()) {
+                        tasks.mark(temp.nextInt() - 1, false);
+                    } else {
+                        throw new DukeException("Please enter an index to unmark.");
+                    }
+                } else if (s.startsWith("delete")) {
+                    Scanner temp = new Scanner(s.substring(6));
+                    if (temp.hasNextInt()) {
+                        tasks.delete(temp.nextInt() - 1);
+                    } else {
+                        throw new DukeException("Please enter an index to delete.");
+                    }
+                } else if (s.startsWith("bye")) {
+                    System.out.println(indent + byePhrase);
+                    break;
+                } else {
+                    throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                }
+            }
```
You should try to avoid deep nesting. Can consider using switch statements instead!

yokester98 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+public abstract class Task {
+    private final String content;
+    private final boolean done;
```
```suggestion
    private final boolean isDone;
```
Naming convention of boolean value should use prefix such as ```is```, ```has```, ```was```, etc.

yokester98 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    public void addTodo(String s) {
```
```suggestion
    public void addTodo(String userInput) {
```
The variable name ```s``` might be too short and not descriptive enough for readers.

yokester98 reviewed (commented):
Overall very clean and readable code! If you have the time, I recommend you work on Javadoc for all your classes and public methods as this will tell the reader how your code works.

Keep it up! 😄

cindyangXQ committed:
Read from file when start up

cindyangXQ committed:
write to file when shut down

cindyangXQ committed:
save Deadline date as LocalDate

cindyangXQ committed:
Merge branch 'branch-Level-8'

cindyangXQ committed:
fix merge

cindyangXQ committed:
file reading bug fix

cindyangXQ committed:
A-MoreOOP

cindyangXQ committed:
Create Package

cindyangXQ committed:
Add JUnit tests

cindyangXQ committed:
Fix bug in printing messages

cindyangXQ committed:
Fix testcases

cindyangXQ committed:
Add JavaDoc

cindyangXQ committed:
Comply with coding standards

cindyangXQ committed:
Solve arrowhead

cindyangXQ committed:
Modify naming

cindyangXQ committed:
Add function "find"

cindyangXQ committed:
Merge branch 'Level-9'

cindyangXQ committed:
Merge branch 'A-CodingStandard'

cindyangXQ committed:
Fix merge

cindyangXQ committed:
Add JavaDoc for Level-9

cindyangXQ committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

cindyangXQ committed:
Fix checkstyle

cindyangXQ committed:
JavaFX tutorial part 1

cindyangXQ committed:
JavaFX tutorial part 2

cindyangXQ committed:
JavaFX tutorial part 3

cindyangXQ committed:
JavaFX tutorial part 4

cindyangXQ committed:
Change images in GUI

cindyangXQ committed:
Adapt Duke for GUI

cindyangXQ committed:
Modify JUnit tests accordingly

cindyangXQ committed:
Refactor code

Refactor code

Code quality requires further improvements.

Let's,
* change the meme that cannot be displayed
* remove unused variables
* rename some variables and methods to make the code more readable
* extract out some methods to avoid arrowheads
* modify some logic to achieve SLAP

cindyangXQ committed:
Add JavaDoc

cindyangXQ committed:
Modify JUnit tests accordingly

cindyangXQ committed:
Add assertions

cindyangXQ committed:
Merge pull request #1 from cindyangXQ/branch-A-Assertions

Add assertions

cindyangXQ committed:
Merge branch 'master' into branch-A-CodeQuality

cindyangXQ committed:
Merge pull request #2 from cindyangXQ/branch-A-CodeQuality

Branch a code quality

cindyangXQ committed:
Add extension C-DetectDuplicates

cindyangXQ committed:
Deal with leading and trailing spaces

cindyangXQ committed:
Bugfix: add deadline date format

cindyangXQ committed:
Add screenshot and user guide

cindyangXQ committed:
Update according to code quality feedback

cindyangXQ committed:
Set theme jekyll-theme-minimal

cindyangXQ committed:
Update README.md

Update README.md

HTML version has wrong indentation in the last section.

Modify to see effect.

cindyangXQ committed:
Set theme jekyll-theme-cayman

cindyangXQ committed:
Modify response message

cindyangXQ committed:
Update GUI

Update GUI

Let's
* align to bottom
* add title for window
* change font

cindyangXQ committed:
Update GUI

Update GUI

Let's
* change font
* add dialog box padding and margin
* add dialog box radius
* add dialog box background color
* make height of dialog box adapt to content

cindyangXQ committed:
Add user guide

cindyangXQ committed:
Update for a new release

cindyangXQ committed:
Update GUI

Update GUI

Texts in the input box and the send button appears to be garbage characters on Apple devices.

Let's change the font.

cindyangXQ committed:
Bugfix

Bugfix

Careless mistake with SceneBuilder.

cindyangXQ committed:
Fix JavaFX version conflict

cindyangXQ committed:
Update README.md
